packages/amazonq/src/app/inline/completion.ts

```diff
@@ -52,71 +52,131 @@ export const CodewhispererInlineCompletionLanguages = [
     { scheme: 'file', language: 'tsx' },
     { scheme: 'file', language: 'jsx' },
 ]
+export class InlineCompletionManager implements Disposable {
+    private disposable: Disposable
+    private inlineCompletionProvider: AmazonQInlineCompletionItemProvider
+    private languageClient: LanguageClient
 
-export function registerInlineCompletion(languageClient: LanguageClient) {
-    const inlineCompletionProvider = new AmazonQInlineCompletionItemProvider(languageClient)
-    languages.registerInlineCompletionItemProvider(CodewhispererInlineCompletionLanguages, inlineCompletionProvider)
+    constructor(languageClient: LanguageClient) {
+        this.languageClient = languageClient
+        this.inlineCompletionProvider = new AmazonQInlineCompletionItemProvider(languageClient)
+        this.disposable = languages.registerInlineCompletionItemProvider(
+            CodewhispererInlineCompletionLanguages,
+            this.inlineCompletionProvider
+        )
+    }
+
+    public dispose(): void {
+        if (this.disposable) {
+            this.disposable.dispose()
+        }
+    }
 
-    const onInlineAcceptance = async (
-        sessionId: string,
-        itemId: string,
-        requestStartTime: number,
-        firstCompletionDisplayLatency?: number
-    ) => {
-        const params: LogInlineCompletionSessionResultsParams = {
-            sessionId: sessionId,
-            completionSessionResult: {
-                [itemId]: {
-                    seen: true,
-                    accepted: true,
-                    discarded: false,
+    public registerInlineCompletion() {
+        const onInlineAcceptance = async (
+            sessionId: string,
+            itemId: string,
+            requestStartTime: number,
+            firstCompletionDisplayLatency?: number
+        ) => {
+            // TODO: also log the seen state for other suggestions in session
+            const params: LogInlineCompletionSessionResultsParams = {
+                sessionId: sessionId,
+                completionSessionResult: {
+                    [itemId]: {
+                        seen: true,
+                        accepted: true,
+                        discarded: false,
+                    },
                 },
-            },
-            totalSessionDisplayTime: Date.now() - requestStartTime,
-            firstCompletionDisplayLatency: firstCompletionDisplayLatency,
+                totalSessionDisplayTime: Date.now() - requestStartTime,
+                firstCompletionDisplayLatency: firstCompletionDisplayLatency,
+            }
+            this.languageClient.sendNotification(logInlineCompletionSessionResultsNotificationType as any, params)
```

Comment:
Is it possible for this to be typed? Is there something in the language-server-runtimes repo that has the type?